bootstrap.go

```diff
@@ -1,147 +1,115 @@
 package coopdatadog
 
 import (
+	"context"
+	"errors"
 	"fmt"
-	"os"
 
-	"github.com/coopnorge/go-datadog-lib/v2/config"
 	"github.com/coopnorge/go-datadog-lib/v2/internal"
-	"github.com/coopnorge/go-logger"
 	datadogLogger "github.com/coopnorge/go-logger/adapter/datadog"
-
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
 	"gopkg.in/DataDog/dd-trace-go.v1/profiler"
 )
 
-// ConnectionType enum type
-type ConnectionType byte
-
-const (
-	// ConnectionTypeSocket sets the connection to Datadog to go through a UNIX socket
-	//
-	// Deprecated: ConnectionTypeSocket. ConnectionTypeAuto should be used.
-	ConnectionTypeSocket ConnectionType = iota
-	// ConnectionTypeHTTP sets the connection to Datadog to go over HTTP
-	//
-	// Deprecated: ConnectionTypeHTTP. ConnectionTypeAuto should be used.
-	ConnectionTypeHTTP
-	// ConnectionTypeAuto sets connection to HTTP or UNIX depending on supplied configuration of DD_TRACE_AGENT_URL
-	ConnectionTypeAuto
-)
-
-// StartDatadog parallel process to collect data for Datadog.
-// connectionType flag related to Datadog connection type, it supports HTTP or socket - values will be used from config.DatadogParameters
-func StartDatadog(cfg config.DatadogParameters, connectionType ConnectionType) error {
+// Start the Datadog integration, use the returned Cancel function to stop the
+// Datadog integration. When calling the StopFunc function traces will be
+// flushed and profiling will be stopped to Datadog.
+//
+// Usage:
+//
+//	package main
+//
+//	import (
+//		"github.com/coopnorge/go-datadog-lib/v2"
+//	)
+//
+//	func main() {
+//		err := run()
+//		if err != nil {
+//			panic(err)
+//		}
+//	}
+//
+//	func run() error {
+//		stop, err := coopdatadog.Start(context.Background())
+//		if err != nil {
+//			return err
+//		}
+//		defer func() {
+//			err := stop()
+//			if err != nil {
+//				panic(err)
+//			}
+//		}()
```

Comment:
since stop() can never return an error, could the signature here be changed to not returning error, so that using it will be easier;
```go
stop, err := coopdatadog.Start(context.Background())
if err != nil {
    return err
}
defer stop
```

Reply:
That is an implementation-detail of the current `stop` function. The next PR (#434) will also start the Metrics-collector as part of the `Start` function, and the metrics-collector can return an error, see: https://github.com/coopnorge/go-datadog-lib/pull/434/files#diff-817f474ba3553c46f0443a1c2f6fad85910778c2deb1ea89466291e0614efc45R124-R128

This is also done to future-proof us, because `trace.Stop()` and `profiler.Stop()` may return errors in the future.